### DISCUSSION TITLE
Sync version updates between pyproject.toml (poetry) and dockerfile with custom registry

### DISCUSSION THREAD

**Discussion by b0g3r** [renovatebot/renovate#31141] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### Please tell us more about your question or problem

I have a python project based on Dockerfile with custom docker registry and using poetry as a package manager. When new python version is released, I want to see one PR, with changes both in pyproject.toml and Dockerfile, but my current setup results in two branches and two PRs:

renovate.json
```json
{
  "enabledManagers": [ "dockerfile", "poetry"],
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true
    },
    {
      "matchUpdateTypes": ["major"],
      "automerge": false
    }
  ]
}
```

Dockerfile
```dockerfile
FROM    us-docker.pkg.dev/my-custom-gar-tf/gar-us/python:3.11
...
```

pyproject.toml
```toml
# ...
[tool.poetry.dependencies]
python = "3.11"
# ...
```

With this configuration renovate finds two dependencies, one with depName `"us-docker.pkg.dev/my-custom-gar-tf/gar-us/python"` and one with depName `"python"`, and therefore creates two distinct MR, both of them would fail on build stage due to incompatibility. Is it possible to sync depNames / updates of two different depNames? I tried to search something with override depName for docker manager, but was unable to find helpful advice

**Comment by secustor** [comment, marked as answer]
Put both of them in the same group using [`groupName`](https://docs.renovatebot.com/configuration-options/#groupname)

**Reply by b0g3r** [reply]
Thanks, @secustor! With a small modification in my pyproject.toml (using tilda specification for python) and grouping both depNames I got it to work with my setup:

renovate.json
```json
{
  "enabledManagers": [ "dockerfile", "poetry"],
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true
    },
    {
      "matchUpdateTypes": ["major"],
      "automerge": false
    },
    {
      "groupName": "python",
      "matchDepNames": [
        "python",
        "us-docker.pkg.dev/my-custom-gar-tf/gar-us/python"
      ],
      "automerge": false
    },
  ]
}
```

pyproject.toml
```toml
# ...
[tool.poetry.dependencies]
python = "~3.11.0"
# ...
```

Patch with changes:
```diff
--- a/Dockerfile
+++ b/Dockerfile

-FROM    us-docker.pkg.dev/my-custom-gar-tf/gar-us/python:3.11
+FROM    us-docker.pkg.dev/my-custom-gar-tf/gar-us/python:3.12

--- a/pyproject.toml
+++ b/pyproject.toml
 
 [tool.poetry.dependencies]
-python = "~3.11.0"
+python = "~3.12.0"

--- a/poetry.lock
+++ b/poetry.lock

 [metadata]
 lock-version = "2.0"
-python-versions = "~3.11.0"
-content-hash = "d3bf8bd1ce145f231eb1c0ec60db2f06967db4a459cda8eea8f8c4fb43b758c3"
+python-versions = "~3.12.0"
+content-hash = "614591417378ad7880aa8a79ce5f87fd82495328ebfcf161914c6190f60967b6"
```
